src/mono/mono/metadata/marshal.h

```diff
@@ -348,6 +348,60 @@ typedef struct {
 	int (*emit_marshal_scalar) (EmitMarshalContext *m, int argnum, MonoType *t, MonoMarshalSpec *spec, int conv_arg, MonoType **conv_arg_type, MarshalAction action);
 } MonoMarshalLightweightCallbacks;
 
+typedef struct {
```

Comment:
This could use a comment.

I wonder if it's worthwhile to structure this with some nested structs (for example group all the method builder stuff), so that it's a little bit easier to understand.

Reply:
Well, I grouped the method builder ones. I am not sure if that actually made it easier to understand?

Reply:
I dunno. feels a bit cleaner to me. although that's mostly because i'm anticipating maybe doing more cleanup here and moving more marshaling stuff into the component.